test/test_parsers.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+
+from cdisutils import parsers
+
+
+@pytest.mark.parametrize(
+    "url, scheme, netloc, bucket, key",
+    (
+        (
+            "s3://ceph.service.consul/bucket/key/name",
+            "s3",
+            "ceph.service.consul",
+            "bucket",
+            "key/name",
+        ), (
+            "s3://s3.amazonaws.com/bucket/key/name",
+            "s3",
+            "s3.amazonaws.com",
+            "bucket",
+            "key/name",
+        ), (
+            "s3://s3.something-here.amazonaws.com/bucket/key/name",
```

Comment:
The `something-here` region has all the latest features, but S3 is three times as expensive.